File: xr18/lib/src/main/java/com/mixer/xr18/lib/data/osc/XR18Addresses.java
package com.mixer.xr18.lib.data.osc;

/**
 * OSC address patterns used by the XR18.
 */
public class XR18Addresses {
    // Discovery
    public static final String XINFO = "/xinfo";
    // Remote subscription (keep-alive)
    public static final String XREMOTE = "/xremote";

    // Channel mix params — ch = 1..16
    public static String chMixFader(int ch)   { return "/ch/" + pad(ch) + "/mix/fader"; }
    public static String chMixOn(int ch)       { return "/ch/" + pad(ch) + "/mix/on"; }
    public static String chMixPan(int ch)      { return "/ch/" + pad(ch) + "/mix/pan"; }

    // Preamp
    public static String headampGain(int ch)  { return "/headamp/" + pad(ch) + "/gain"; }

    // EQ
    public static String chEqOn(int ch)       { return "/ch/" + pad(ch) + "/eq/on"; }
    public static String chEqBandG(int ch, int band) { return "/ch/" + pad(ch) + "/eq/" + band + "/g"; }

    private static String pad(int ch) {
        return (ch < 10) ? ("0" + ch) : String.valueOf(ch);
    }
}

File: xr18/lib/src/main/java/com/mixer/xr18/lib/data/osc/XR18Addresses.kt
package com.mixer.xr18.lib.data.osc

/**
 * OSC address patterns used by the XR18.
 */
object XR18Addresses {
    // Discovery
    const val XINFO = "/xinfo"

    // Remote subscription (keep-alive)
    const val XREMOTE = "/xremote"

    // Channel mix params — xx = 01..16
    fun chMixFader(ch: Int)   = "/ch/$ch/mix/fader"
    fun chMixOn(ch: Int)      = "/ch/$ch/mix/on"
    fun chMixPan(ch: Int)     = "/ch/$ch/mix/pan"

    // Preamp
    fun headampGain(ch: Int)  = "/headamp/$ch/gain"

    // EQ
    fun chEqOn(ch: Int)       = "/ch/$ch/eq/on"
    fun chEqBandG(ch: Int, band: Int) = "/ch/$ch/eq/$band/g"
}
